plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -2309,6 +2344,76 @@ public void finishStatisticsCollection(ConnectorSession session, ConnectorTableH
                 computedStatistics);
     }
 
+    private void generateMissingStatistics(ConnectorSession session, DeltaLakeTableHandle tableHandle, Collection<ComputedStatistics> computedStatistics)
+    {
+        Map<String, AddFileEntry> addFileEntriesWithNoStats = metastore.getValidDataFiles(tableHandle.getSchemaTableName(), session).stream()
+                .filter(addFileEntry -> addFileEntry.getStats().isEmpty()
+                        || addFileEntry.getStats().get().getNumRecords().isEmpty()
+                        || addFileEntry.getStats().get().getMaxValues().isEmpty()
+                        || addFileEntry.getStats().get().getMinValues().isEmpty()
+                        || addFileEntry.getStats().get().getNullCount().isEmpty())
+                // Statistics returns whole path to file, so we need to create corresponding map key for AddFileEntry.
+                .map(addFileEntry -> Map.entry(appendPath(tableHandle.getLocation(), addFileEntry.getPath()), addFileEntry))
+                .collect(Collectors.toUnmodifiableMap(Entry::getKey, Entry::getValue));
+
+        if (addFileEntriesWithNoStats.isEmpty()) {
+            return;
+        }
+
+        List<String> partitionColumns = getPartitionedBy(getTableMetadata(session, tableHandle).getProperties());
+        Map<String, DeltaLakeColumnMetadata> columnsMetadata = DeltaLakeSchemaSupport.extractSchema(tableHandle.getMetadataEntry(), typeManager).stream()
+                .filter(column -> !partitionColumns.contains(column.getName()) && !column.getColumnMetadata().isHidden())
```

Comment:
See here that the check is made in a case insensitive fashion

https://github.com/trinodb/trino/blob/8286f99980438a2000e0f9b6c738d920b608c682/plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java#L2744

Reply:
alternative for obtaining the regular columns (check against `REGULAR`)

https://github.com/trinodb/trino/blob/8286f99980438a2000e0f9b6c738d920b608c682/plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java#L2744